modules/core/site.js

```diff
@@ -1573,6 +1575,19 @@ var Hm_Utils = {
         return res;
     },
 
+    seach_from_local_storage: function(pattern) {
```

Comment:
Search is misspelled here. It seems rather inefficient to look through all the entries in sessionStorage. Also, save_to_local_storage checks for Storage != void(0) to prevent other JS errors, we should do the same here, I think.

What is the purpose of this function? You seem to load all the stored imap source keys in the storage and save back with value of 1. Not sure I understand the need to do this. Do you invalidate some kind of caching?

Reply:
> Do you invalidate some kind of caching?

Yes exactly, when I access the message page a cache like: 62_imap_xxxxxxxx_xxxx is created. The problem is that the next time when I open this email again it will be retrieved from the cache so if I add a label I won't be able to get it updated. This is why I added a function that allows me to search the cache for anything that has this format: int_imap_xxxxxxxx_xxxx and delete them